frontend/src/views/producer/producer-modal/index.js

```diff
@@ -0,0 +1,103 @@
+import React, { Component } from 'react';
+import Modal, { Content } from '../../../components/modal';
+import Typography from '../../../components/typography';
+import { connect } from '../../../utils';
+import { withTheme } from 'styled-components';
+import Icon from 'antd/lib/icon';
+
+export default withTheme(
+  connect('store')(
+    class AppModal extends Component {
+      state = {
+        show: false,
+        handledSignalToShow: false,
+      };
+      constructor(props) {
+        super(props);
+        if (props.store.ui.auth.validateTokenFailed) {
+          this.state = { show: true };
+          window.setTimeout(() => {
+            try {
+              props.store.ui.auth.clearValidationError();
+            } catch (error) {
+              console.error(error);
+            }
+          }, 5000);
+        }
+      }
+      componentWillUpdate() {
+        // only show modal if its 0.5s pass waiting.
+        // this is to avoid flashing effects
+        if (
+          this.props.store.ui.auth.validateTokenInProgress &&
+          !this.state.handledSignalToShow
+        ) {
+          this.setState(
+            {
+              handledSignalToShow: true,
+            },
+            () => {
+              new Promise((resolve, reject) => {
+                window.setTimeout(resolve, 300);
+              }).then(() => {
+                this.setState({
+                  show: true,
+                });
+              });
+            },
+          );
+        }
+        if (this.props.store.ui.auth.validateTokenFailed) {
+          window.setTimeout(() => {
+            try {
+              this.props.store.ui.auth.clearValidationError();
+            } catch (error) {
+              console.error(error);
+            }
+          }, 5000);
+        }
+      }
+      render() {
+        const { auth } = this.props.store.ui;
+        const { palette } = this.props.theme;
+
+        const showSignalFromStore =
+          auth.validateTokenInProgress || auth.validateTokenFailed;
+        return (
+          <Modal
+            show={showSignalFromStore && this.state.show}
```

Comment:
The visibility of the modal is now controlled by three separate values which are evaluated in different places. I would rather just have one value (eg. `this.state.show`) to control the view and have the logic to set it elsewhere (not in render)

Reply:
i wouldnt want to pollute the global state with very component specific logics like defer showing after 0.5s to avoid flashing effects. I think this is one thing I saw redux-saga people do.
So much for `pulling business logic out of rendering logic` when you start bleeding rendering logic into your app logic :D

Reply:
I geniunely think it is fine to enhance state from store as long as you keep it predictable.
After all it is about the predictability that keeps bug out of a program.

Reply:
Yes, I understand your views. One improvement you could do though is to include the `this.state.show` to the condition above it instead of having it separately there.

`const showSignalFromStore = (auth.validateTokenInProgress || auth.validateTokenFailed) && this.state.show;`

That way the JSX keeps cleaner and readability increases.

Reply:
i get your point. i can revisit this pr after i make a pr about filter